### PR TITLE
Add "additional Medicare tax" to list of MO itemized deductions

### LINKED ISSUE #1808: Incomplete list of MO itemized deductions
The list of itemized deductions available under the MO income tax is incomplete in the 0.219.0 version.   The Itemized Deductions Worksheet on the [2021 MO Form 1040](https://dor.mo.gov/forms/MO-1040%20Fillable%20Calculating_2021.pdf) contains the following:

Screenshot 2023-02-20 at 8 09 04 AM

Notice that Line 6 directs taxpayers to the [instructions](https://dor.mo.gov/forms/MO-1040%20Instructions_2021.pdf#page=16), which contain this:

Screenshot 2023-02-20 at 8 14 26 AM

The list of MO itemized deductions (in the current version of `parameters/gov/states/mo/tax/income/deductions/itemized.yaml`) does not include the "additional Medicare tax".

### PR DESCRIPTION
**Pull request by martinholmer**
Fixes #1808.

The code coverage percent goes down slightly with this pull request because the code streamlining reduces the number of Python statements by three.   This pull request does not introduce any untested Python statements.

### CODE REVIEW

File: policyengine_us/variables/gov/states/mo/tax/income/deductions/mo_itemized_deductions.py [hunk 1]
```diff
@@ -15,12 +15,8 @@ class mo_itemized_deductions(Variable):
...
     def formula(tax_unit, period, parameters):
-        total_itemized_federal_deductions = parameters(
-            period
-        ).gov.states.mo.tax.income.deductions.itemized
-        deductions = [
-            deduction for deduction in total_itemized_federal_deductions
-        ]
+        p = parameters(period).gov.states.mo.tax.income
+        deductions = [deduction for deduction in p.deductions.itemized]
         deduction_value = add(tax_unit, period, deductions)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/mo/tax/income/deductions/mo_itemized_deductions.py:20]:
```suggestion
        deduction_value = add(tax_unit, period, p.deductions.itemized)
```

File: policyengine_us/parameters/gov/irs/deductions/itemized_deductions.yaml [hunk 2]
```diff
@@ -8,5 +8,5 @@ values:
     - casualty_loss_deduction
 metadata:
   unit: variable
-  name: taxable_income_deductions_if_itemizing
-  label: Taxable income deductions if itemizing
+  name: itemized_deductions
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/irs/deductions/itemized_deductions.yaml:11]:
```suggestion
```

File: policyengine_us/variables/gov/states/mo/tax/income/deductions/mo_itemized_deductions.py [hunk 3]
```diff
@@ -15,12 +15,8 @@ class mo_itemized_deductions(Variable):
...
 
     def formula(tax_unit, period, parameters):
-        total_itemized_federal_deductions = parameters(
-            period
-        ).gov.states.mo.tax.income.deductions.itemized
-        deductions = [
-            deduction for deduction in total_itemized_federal_deductions
-        ]
+        p = parameters(period).gov.states.mo.tax.income
+        deductions = [deduction for deduction in p.deductions.itemized]
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/mo/tax/income/deductions/mo_itemized_deductions.py:19]:
```suggestion
```

**Review by MaxGhenis — APPROVED** [review]

**Comment by martinholmer** [comment]
@MaxGhenis,  Thanks for the additional code simplification.

**Pull request merged**